File: frontend/lib/models/user_response.dart
class UserResponse {
  final int id;
  final String teamName;
  final String userName;
  final String role;

  UserResponse({
    required this.id,
    required this.teamName,
    required this.userName,
    required this.role,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      id: json['id'] as int,
      teamName: json['teamName'] as String,
      userName: json['userName'] as String,
      role: json['role'] as String,
    );
  }
}

File: backend/src/main/java/com/roccia/backend/dto/response/UserResponse.java
package com.roccia.backend.dto.response;

import com.roccia.backend.domain.Role;
import com.roccia.backend.domain.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponse {
    private Long id;
    private String teamName;
    private String userName;
    private Role role;

    public static UserResponse from(User user) {
        return UserResponse.builder()
                .id(user.getId())
                .teamName(user.getTeam().getName())
                .userName(user.getUserName())
                .role(user.getRole())
                .build();
    }
}
